x-pack/plugins/enterprise_search/public/applications/workplace_search/components/overview/overview_logic.ts

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { HttpSetup } from 'src/core/public';
+
+import { storeLogic } from '../../../shared/store';
+
+import { IAccount, IFlashMessagesProps, IOrganization } from '../../types';
+
+import { IFeedActivity } from './recent_activity';
+
+export interface IOverviewServerData {
+  hasUsers: boolean;
+  hasOrgSources: boolean;
+  canCreateContentSources: boolean;
+  canCreateInvitations: boolean;
+  isOldAccount: boolean;
+  sourcesCount: number;
+  pendingInvitationsCount: number;
+  accountsCount: number;
+  personalSourcesCount: number;
+  activityFeed: IFeedActivity[];
+  organization: IOrganization;
+  isFederatedAuth: boolean;
+  currentUser: {
+    firstName: string;
+    email: string;
+    name: string;
+    color: string;
+  };
+  fpAccount: IAccount;
+}
+
+export interface IOverviewActions {
+  setServerData(serverData: IOverviewServerData): void;
+  setFlashMessages(flashMessages: IFlashMessagesProps): void;
+  setHasErrorConnecting(hasErrorConnecting: boolean): void;
+  initializeOverview({ http }: { http: HttpSetup }): void;
+}
+
+export interface IOverviewValues extends IOverviewServerData {
+  dataLoading: boolean;
+  hideOnboarding: boolean;
+  hasErrorConnecting: boolean;
+  flashMessages: IFlashMessagesProps;
+}
+
+interface IListenerParams {
+  actions: IOverviewActions;
+}
+
+export const OverviewLogic = storeLogic({
+  actions: (): IOverviewActions => ({
+    setServerData: (serverData: IOverviewServerData) => serverData,
+    setFlashMessages: (flashMessages: IFlashMessagesProps) => ({
+      flashMessages,
+    }),
+    setHasErrorConnecting: (hasErrorConnecting: boolean) => ({ hasErrorConnecting }),
+    initializeOverview: ({ http }: { http: HttpSetup }) => ({ http }),
+  }),
+  reducers: () => ({
+    organization: [
+      {},
+      {
```

Comment:
woof haha, prettier has _opinions_ about our old formatting 😭

Reply:
Yeah I really don't like that, nor do I like having to litter everything with the interfaces either
```
setServerData: (_: IOverviewValues, { fpAccount }: IOverviewValues) => fpAccount,
```

Reply:
Oh yeah, wtf 😞 That does feel kinda bad... maybe this is one of those things that will get better when Kea's author implements Typescript support?